Limit path search within range

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #164738
* __->__ #164581
* #164569
* #164568


When we are looking if two nodes are dependent, limit path search within the bounds of their node idxs. 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

eellison committed:
Limit path search within range

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/fx_passes/overlap_preserving_bucketer.py:
```diff
@@ -11,10 +12,58 @@
 from torch.utils._ordered_set import OrderedSet
 
 
+@dataclass
```
Nit: might want to make slots=true here since the class looks potentially performance sensitive/ in hot paths and only modifies declared instance variables

eellison committed:
Update on "Limit path search within range"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

eellison committed:
Update on "Limit path search within range"



When we are looking if two nodes are dependent, limit bath search within the bounds of their node idxs. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

ezyang reviewed (approved):
High level approach seems fine, I didn't do a fine grained code review

eellison committed:
Update on "Limit path search within range"



When we are looking if two nodes are dependent, limit path search within the bounds of their node idxs. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]
